src/Doctrine/DoctrineHelper.php

```diff
@@ -265,4 +297,36 @@ public function isKeyword(string $name): bool
 
         return $connection->getDatabasePlatform()->getReservedKeywordsList()->isKeyword($name);
     }
+
+    /**
+     * this method tries to find the correct MappingDriver for the given namespace/class
+     * To determine which MappingDriver belongs to the class we check the prefixes configured in Doctrine and use the
+     * prefix that has the closest match to the given $namespace.
+     *
+     * this helper function is needed to create entities with the configuration of doctrine if they are not yet been registered
+     * in the ManagerRegistry
+     */
+    public function getMappingDriverForNamespace(string $namespace): ?MappingDriver
```

Comment:
I think this method can be private. It IS used in `MakeEntity`. However, I think it could be replaced by `-> doesClassUsesAttributes()` and then this could be private.

Well, this method is also used to populate the `$mappingDriver` variable in `MakeEntity`, which is then passed to `MakeEntity::createClassManipulator()`. But, I think instead we should pass a `string $className` to `createClassManipulator` and then it can call methods on `DoctrineHelper` to determine if attributes should be used.

Reply:
possible solution implemented. needs testing